### PR TITLE
Docs nav

### LINKED ISSUE #1615: Implement docs sidebar
See https://cloud.githubusercontent.com/assets/3925912/25151697/7f057428-2454-11e7-95e7-541e6d7c2abc.png for most up-to-date mockup of it.

### LINKED ISSUE #1668: Strip and reformat html from yaml
this includes:

- object links
- ul li in descriptions

### PR DESCRIPTION
**Pull request by mparke**
closes #1615 
closes #1668

### CODE REVIEW

File: docs/package.json [hunk 1]
```diff
@@ -56,7 +56,7 @@
     "lodash": "^4.16.4",
     "moment": "^2.17.1",
     "moment-timezone": "^0.5.11",
-    "react": "^15.1.0",
+    "react": "https://registry.npmjs.org/react/-/react-15.5.4.tgz",
```
Comment (eatonphil) [inline docs/package.json]: I'm getting deja vu... Why are we doing this?

File: docs/src/components/Endpoint.js [hunk 2]
```diff
@@ -10,9 +10,9 @@ import { default as Method } from './Method';
 
 export default function Endpoint(props) {
   const { route } = props;
-  const { endpoint } = route;
-  const { crumbs, description, methods, path, resourceObject } = endpoint;
-
+  const { crumbs, endpoint } = route;
+  const { description, methods, path, resource } = endpoint;
+  console.log(description);
```
Comment (eatonphil) [inline docs/src/components/Endpoint.js]: console.log?

**Comment by eatonphil** [comment]
Ah crap. Could you add the docs and components folders to the lint script in the manager's package.json. And then could you address the linter issues?

File: docs/package.json (hunk 1 above)
Comment (mparke) [inline docs/package.json]: damn, I don't know why this changed to this, will test with --save to see if it's us or something else

**Comment by mparke** [comment]
Follow up for linting:

https://github.com/linode/manager/issues/1699

File: docs/src/data/endpoints/index.js [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+// export { default as regions } from './datacenters.js';
+// export { default as distributions } from './distributions.js';
+// export { default as domains } from './dnszones.js';
+// export { default as events } from './events.js';
+// export { default as kernels } from './kernels.js';
+// export { default as linodes } from './linodes.js';
+// export { default as networking } from './networking.js';
+// export { default as services } from './services.js';
+// export { default as stackscripts } from './stackscripts.js';
+// export { default as supporttickets } from './supporttickets.js';
```
Comment (na3d) [inline docs/src/data/endpoints/index.js]: Do we still need this commented out code?

File: docs/src/data/objects/datacenter.js [hunk 4]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+module.exports = {"name":"Datacenter","prefix":"dctr","description":"Datacenter objects describe the datacenters available for Linode services.\n","schema":{"id":{"_type":"string","_value":"newark"},"label":{"_type":"string","_value":"Newark, NJ","_description":"Human-friendly datacenter name.","_filterable":true},"country":{"_type":"string","_value":"US","_description":"Country","_filterable":true}},"enums":{"Datacenter":{"dallas":"Dallas, TX","fremont":"Fremont, CA","atlanta":"Atlanta, GA","newark":"Newark, NJ","london":"London, England, UK","singapore":"Singapore, SG","frankfurt":"Frankfurt, DE"}}};
```
Comment (na3d) [inline docs/src/data/objects/datacenter.js]: Should this still exist?

File: docs/src/data/objects/index.js [hunk 5]
```diff
@@ -29,6 +33,8 @@ module.exports.apiObjectMap = {
   datacenter,
```
Comment (na3d) [inline docs/src/data/objects/index.js]: isn't this now regions?

File: docs/src/data/objects/datacenter.js (hunk 4 above)
Comment (mparke) [inline docs/src/data/objects/datacenter.js]: Yes, all of the object files are still used right now.

File: docs/src/data/objects/index.js (hunk 5 above)
Comment (mparke) [inline docs/src/data/objects/index.js]: yes, good catch

File: docs/src/data/endpoints/index.js (hunk 3 above)
Comment (mparke) [inline docs/src/data/endpoints/index.js]: nope, will remove

File: docs/src/data/objects/index.js [hunk 6]
```diff
@@ -1,10 +1,11 @@
 const backup = require('./backup');
 const backupsresponse = require('./backupsresponse');
-const datacenter = require('./datacenter');
 const disk = require('./disk');
 const distribution = require('./distribution');
 const dnszone = require('./dnszone');
```
Comment (na3d) [inline docs/src/data/objects/index.js]: Do we still need dnszone stuff, since we have domain now?

**Pull request merged**